### PR TITLE
change the ribbon example into a tracer example.

### PR DESCRIPTION
**Pull request by naasblod**
I think this could be more helpful for people wanting to use the ribbon example. In my mind "tracers" are the most used application for this sort of thing, If you feel it isn't just close it.

run with:
 `cargo run --example ribbon --features="bevy/bevy_winit bevy/bevy_pbr 3d bevy/bevy_gizmos"`

### CODE REVIEW

File: examples/ribbon.rs [hunk 1]
```diff
@@ -125,9 +124,26 @@ fn setup(mut commands: Commands, mut effects: ResMut<Assets<EffectAsset>>) {
...
     for mut transform in query.iter_mut() {
-        transform.translation = vec3(f32::cos(theta), f32::sin(theta), 0.0) * 5.0;
+        let time = timer.elapsed_seconds() * TIME_SCALE;
+        let pos = vec3(
+            (1.0 - K) * (time.clone().cos()) + (L * K) * (((1.0 - K) / K) * time.clone()).cos(),
+            (1.0 - K) * (time.clone().sin()) - (L * K) * (((1.0 - K) / K) * time.clone()).sin(),
+            0.0,
+        ) * SHAPE_SCALE;
+
+        //let pos = vec3(f32::cos(theta), f32::sin(theta), 0.0) * 5.0;
```
Comment (djeedai) [inline examples/ribbon.rs:144]: Delete this?

**Review by djeedai — COMMENTED** [review]
If I understand correctly you're proposing to replace the purely GPU driven motion with a property-based one, to show how a ribbon can follow a target as a kind of trail, is that it?

I don't have any objection, but if you feel that tracers are common (I don't have an opinion, probably that's correct) then I'd expand a tiny bit the top level documentation to mention that this example shows a "tracer" (so that if someone search that term they find that example) and explain how a property is used to link the particle position to some CPU position (as this is probably a common feature to want).

File: examples/ribbon.rs (hunk 1 above)
Comment (naasblod) [reply examples/ribbon.rs:144]: Thanks for the comments I think those are great additions!

> replace the purely GPU driven motion with a property-based one
Yeah that's exactly it.

If you want I could expand on the example and make it separate to the ribbon if you feel like both are useful.
Meanwhile I'll crack on with the changes proposed here.

**Comment by naasblod** [comment]
https://github.com/user-attachments/assets/82454667-9264-444a-ae42-5427b666a5f8

**Comment by naasblod** [comment]
Is there anything else you'd like me to do with this or should I close it? Maybe move to a separate example?

**Comment by pcwalton** [comment]
This is fine by me. Note that the little gaps between ribbon segments are something I've been thinking about fixing with the singly linked list experiment (that needs #387 to land first).

**Comment by djeedai** [comment]
Apology @naasblod I had to prioritize merging the large fix from @pcwalton for ribbons due to its size and criticality, which now makes this change conflict. Please let me know if you have time ideally to fix, otherwise I'll try to find time to do it myself. Thanks for your understanding!

**Comment by naasblod** [comment]
No worries friend ❤️! I'll take care of it!

**Comment by naasblod** [comment]
I think that does it. Could use a final review

**Comment by djeedai** [comment]
@naasblod the example fails to build:

```
error[E0412]: cannot find type `Gizmos` in this scope
   --> examples\ribbon.rs:115:17
    |
115 |     mut gizmos: Gizmos,
    |                 ^^^^^^ not found in this scope
```

**Comment by naasblod** [comment]
@djeedai should be fixed now.

**Review by djeedai — APPROVED** [review]

**Comment by djeedai** [comment]
@naasblod for next time, if you need a dependency for examples and tests only, you can add it to the `[dev-dependencies]` section in `Cargo.toml`.

**Comment by naasblod** [comment]
Ah! Thanks, that's useful to know!

**Pull request merged**